PhysicsTools/PatAlgos/python/producersHeavyIons/heavyIonJetSetup.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import division
+
+from RecoHI.HiJetAlgos.HiRecoPFJets_cff import akCs4PFJets
+from RecoHI.HiJetAlgos.HiGenJets_cff import ak5HiGenJets
+from RecoHI.HiJetAlgos.HiGenCleaner_cff import heavyIonCleanedGenJets
+from PhysicsTools.PatAlgos.producersHeavyIons.heavyIonJets_cff import *
+from PhysicsTools.PatAlgos.tools.helpers import *
+
+def setupHeavyIonJets(process, tag, radius, task):
+    addToProcessAndTask(
+        tag + 'Jets',
+        akCs4PFJets.clone(rParam = radius / 10),
+        process,
+        task)
+
+    genjetcollection = 'ak' + str(radius) + 'HiGenJets'
+
+    addToProcessAndTask(
+        genjetcollection,
+        ak5HiGenJets.clone(rParam = radius / 10),
+        process,
+        task)
+
+    addToProcessAndTask(
+        'ak' + str(radius) + 'HiCleanedGenJets',
+        heavyIonCleanedGenJets.clone(src = genjetcollection),
+        process,
+        task)
+
+    from PhysicsTools.PatAlgos.tools.jetTools import addJetCollection
+
+    addJetCollection(
+        process,
+        labelName = 'AKCs4PF',
+        jetSource = cms.InputTag('akCs4PFJets'),
+        btagDiscriminators = [
+            'simpleSecondaryVertexHighEffBJetTags',
+            'simpleSecondaryVertexHighPurBJetTags',
+            'combinedSecondaryVertexV2BJetTags',
+            'jetBProbabilityBJetTags',
+            'trackCountingHighEffBJetTags',
+            'trackCountingHighPurBJetTags'
+        ],
+        genJetCollection = cms.InputTag('ak4HiCleanedGenJets'), 
+        jetCorrections = ('AK4PF', ['L2Relative'], 'None'),
+        getJetMCFlavour = False, # jet flavor disabled
+    )
+
+    process.patJetsAKCs4PF.addAssociatedTracks = True
+    process.patJetsAKCs4PF.addBTagInfo = True
+    process.patJetsAKCs4PF.useLegacyJetMCFlavour = True
+    process.patJetCorrFactorsAKCs4PF.useNPV = False
+    process.patJetCorrFactorsAKCs4PF.useRho = False
+    process.patJetPartonMatchAKCs4PF.matched = "cleanedPartons"
+
+
+def aliasCsJets(process, tag):
+    delattr(process, 'patJets')
+
+    source = cms.VPSet(cms.PSet(type = cms.string('patJets')))
+    process.patJets = cms.EDAlias(**{ 'patJets' + tag: source })
```

Comment:
I'm not sure why it was done this way (the original author of this code in our private repo left CMS).  A priori it certainly seems more logical to simply update the input to pat jets.  Why do you propose instead to keep separate collections if the default patJets is not useful for heavy ions?

Reply:
> I'm not sure why it was done this way (the original author of this code in our private repo left CMS). A priori it certainly seems more logical to simply update the input to pat jets. Why do you propose instead to keep separate collections if the default patJets is not useful for heavy ions?

mostly because I still didn't figure out which parts of the defaults you are disabling or will need to remake structurally (considering everything else that's built on top of the default patJets).
Otherwise, a straightforward update to patJets inputs would be just fine.

Reply:
I updated the code to simply overwrite patJets at the same place the aliasing was previously done.   I'm not sure this is what you had in mind, but it achieves the desired effect.